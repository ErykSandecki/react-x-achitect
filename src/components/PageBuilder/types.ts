// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
