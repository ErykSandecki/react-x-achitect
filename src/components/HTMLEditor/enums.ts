export enum BlockHeight {
  small = 50,
  medium = 70,
  large = 100,
}

export enum BlockWidth {
  small = 50,
  medium = 200,
  large = 350,
}

export enum MouseMode {
  comment = 'comment',
  default = 'default',
  menu = 'menu',
  move = 'move',
}
