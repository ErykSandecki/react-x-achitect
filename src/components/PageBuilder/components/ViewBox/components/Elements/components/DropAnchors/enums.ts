export enum DropAnchorsPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
