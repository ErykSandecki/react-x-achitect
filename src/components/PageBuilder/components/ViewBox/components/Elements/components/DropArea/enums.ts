export enum DropAreaPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
