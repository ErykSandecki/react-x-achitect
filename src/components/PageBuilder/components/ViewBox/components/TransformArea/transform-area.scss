@import '../../../../styles/z-indexes.scss';

.TransformArea {
  height: 100%;
  left: 0;
  pointer-events: none;
  position: absolute;
  top: 0;
  width: 100%;

  &__anchor {
    $--size: 5px;
    pointer-events: all;
    position: absolute;

    &--east {
      cursor: ew-resize;
      height: calc(100% - #{$--size});
      right: calc(-#{$--size} / 2);
      top: calc(#{$--size} / 2);
      width: $--size;
    }

    &--none {
      display: none;
    }

    &--north {
      cursor: ns-resize;
      height: $--size;
      left: calc(#{$--size} / 2);
      top: calc(-#{$--size} / 2);
      width: calc(100% - #{$--size});
    }

    &--north-east {
      cursor: nesw-resize;
      height: $--size;
      right: calc(-#{$--size} / 2);
      top: calc(-#{$--size} / 2);
      width: $--size;
    }

    &--north-west {
      cursor: nwse-resize;
      height: $--size;
      left: calc(-#{$--size} / 2);
      top: calc(-#{$--size} / 2);
      width: $--size;
      z-index: 10;
    }

    &--south {
      cursor: ns-resize;
      height: $--size;
      left: calc(#{$--size} / 2);
      bottom: calc(-#{$--size} / 2);
      width: calc(100% - #{$--size});
    }

    &--south-east {
      cursor: nwse-resize;
      height: $--size;
      right: calc(-#{$--size} / 2);
      bottom: calc(-#{$--size} / 2);
      width: $--size;
    }

    &--south-west {
      cursor: nesw-resize;
      height: $--size;
      left: calc(-#{$--size} / 2);
      bottom: calc(-#{$--size} / 2);
      width: $--size;
    }

    &--west {
      cursor: ew-resize;
      height: calc(100% - #{$--size});
      left: calc(-#{$--size} / 2);
      top: calc(#{$--size} / 2);
      width: $--size;
    }
  }
}
