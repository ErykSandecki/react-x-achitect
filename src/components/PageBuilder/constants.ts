export const translationNameSpace = 'pageBuilder';

export const TOOLBAR_HEIGHT = 56;
