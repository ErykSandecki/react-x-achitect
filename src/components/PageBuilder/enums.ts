export enum Constrain {
  bottom = 'bottom',
  centerHorizontal = 'centerHorizontal',
  centerVertical = 'centerVertical',
  left = 'left',
  right = 'right',
  top = 'top',
}
