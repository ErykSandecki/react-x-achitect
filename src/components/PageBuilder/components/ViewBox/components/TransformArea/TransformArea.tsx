import { FC } from 'react';

// hooks
import { useTransformAreaEvents } from './hooks/useTransformAreaEvents';
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// types
import { Anchor } from 'store/pageBuilder/enums';
import { MouseMode } from 'components/PageBuilder/enums';
import { TElement } from 'types';

// styles
import styles from './transform-area.scss';

// utils
import { enumToArray } from 'utils';

export type TResizeAreaProps = {
  height: TElement['height'];
  id: TElement['id'];
  moseMode: MouseMode;
  width: TElement['width'];
  x: TElement['positionAbsolute']['x'];
  y: TElement['positionAbsolute']['y'];
};

const TransformArea: FC<TResizeAreaProps> = ({
  height,
  id,
  moseMode,
  width,
  x,
  y,
}) => {
  const events = useTransformAreaEvents(height, id, moseMode, width, x, y);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <div className={cx(classNamesWithTheme[className])}>
      {enumToArray(Anchor).map((anchor) => (
        <div
          className={cx(
            classNamesWithTheme.anchor.name,
            classNamesWithTheme.anchor.modificators[
              anchor as keyof typeof Anchor
            ],
          )}
          key={anchor as keyof typeof Anchor}
          onMouseDown={(event) => events.onMouseDown(anchor as Anchor, event)}
        />
      ))}
    </div>
  );
};

export default TransformArea;
