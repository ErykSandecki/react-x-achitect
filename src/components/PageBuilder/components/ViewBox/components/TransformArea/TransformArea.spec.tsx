import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import TransformArea from './TransformArea';

// mocks
import { selectedElementMock } from 'test/mocks/reducer/pageBuilderMock';
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// types
import { MouseMode } from 'components/PageBuilder/enums';

describe('TransformArea snapshots', () => {
  it('should render TransformArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <TransformArea
          height={100}
          id={selectedElementMock.id}
          moseMode={MouseMode.default}
          width={100}
          x={100}
          y={100}
        />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
