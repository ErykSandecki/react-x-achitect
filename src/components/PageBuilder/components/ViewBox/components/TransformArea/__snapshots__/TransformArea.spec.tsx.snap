// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformArea snapshots should render TransformArea 1`] = `
<DocumentFragment>
  <div
    class="TransformArea TransformArea--light"
  >
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--east TransformArea__anchor--east--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--none TransformArea__anchor--none--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--north TransformArea__anchor--north--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--northEast TransformArea__anchor--northEast--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--northWest TransformArea__anchor--northWest--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--south TransformArea__anchor--south--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--southEast TransformArea__anchor--southEast--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--southWest TransformArea__anchor--southWest--light"
    />
    <div
      class="TransformArea__anchor TransformArea__anchor--light TransformArea__anchor--west TransformArea__anchor--west--light"
    />
  </div>
</DocumentFragment>
`;
